src/forward.ts

```diff
@@ -39,13 +49,13 @@ const forward = async (request, response, handlerOpts) => new Promise(async (res
         }
     }
 
-    const fn = origin.startsWith('https:') ? https.request : http.request;
+    const fn = origin!.startsWith('https:') ? https.request : http.request;
 
-    const client = fn(origin, options, async (clientResponse) => {
+    const client = fn(origin!, options as unknown as http.ClientRequestArgs, async (clientResponse) => {
         try {
             // This is necessary to prevent Node.js throwing an error
             let { statusCode } = clientResponse;
```

Comment:
we could handle the optionallity right here to get rid of all the `!` in following lines. either via casting `clientResponse` or by changing this to `let statusCode = clientResponse.statusCode!`

(personally i am not in the "prefer object destructing everywhere blindly" :P)